packages/page-staking/src/Performance/Performance.tsx

```diff
@@ -0,0 +1,321 @@
+  currentSessionMode: boolean,
+}
+
+interface CurrentEraValidators extends Struct {
+  reserved: Vec<AccountId>;
+  nonReserved: Vec<AccountId>;
+}
+
+interface CommitteeSize extends Struct {
+  nonReservedSeats: u32
+  reservedSeats: u32,
+}
+
+type SessionIndexEntry = [{ args: [EraIndex] }, Option<u32>];
+
+function Performance ({ className = '', currentSessionMode, favorites, sessionEra, targets, toggleFavorite }: Props): React.ReactElement<Props> {
+  const { api } = useApi();
+
+  const erasStartSessionIndex = useCall<SessionIndexEntry[]>(api.query.staking.erasStartSessionIndex.entries);
+  const [firstBlockInSessionHash, setFirstBlockInSessionHash] = useState<Hash | null>(null);
+  const [lastBlockInSessionHash, setLastBlockInSessionHash] = useState<Hash | null>(null);
+  const [currentEraValidators, setCurrentEraValidators] = useState<CurrentEraValidators | null>(null);
+  const [committeeSize, setCommitteeSize] = useState<CommitteeSize | null>(null);
+  const [sessionValidatorBlockCountLookup, setSessionValidatorBlockCountLookup] = useState<Record<string, number>>({});
+  const [firstSessionBlockAuthor, setFirstSessionBlockAuthor] = useState<string>('');
+  const [isPalletElectionsSupported, setIsPalletElectionsSupported] = useState<boolean>(false);
+
+  const MINIMUM_SUPPORTED_ELECTIONS_PALLET_VERSION = 3;
+
+  const era: number | null = useMemo(() => {
+    if (currentSessionMode) {
+      if (!sessionEra.era) {
+        console.warn('sessionEra.era should not be undefined in the current session mode!');
+
+        return null;
+      }
+
+      return sessionEra.era;
+    }
+
+    if (!erasStartSessionIndex) {
+      return null;
+    }
+
+    const erasStartSessionIndexLookup: [number, number][] = [];
+
+    erasStartSessionIndex.filter(([, values]) => values.isSome)
+      .forEach(([key, values]) => {
+        const eraIndex = key.args[0];
+
+        erasStartSessionIndexLookup.push([eraIndex.toNumber(), values.unwrap().toNumber()]);
+      });
+    erasStartSessionIndexLookup.sort(([eraIndexA], [eraIndexB]) => {
+      return eraIndexA - eraIndexB;
+    });
+
+    for (let i = 0; i < erasStartSessionIndexLookup.length; i++) {
+      const eraIndex = erasStartSessionIndexLookup[i][0];
+      const currentEraSessionStart = erasStartSessionIndexLookup[i][1];
+      const currentEraSessionEnd = i + 1 < erasStartSessionIndexLookup.length ? erasStartSessionIndexLookup[i + 1][1] - 1 : null;
+
+      if (currentEraSessionStart <= sessionEra.session && currentEraSessionEnd && sessionEra.session <= currentEraSessionEnd) {
+        return eraIndex;
+      }
+    }
+
+    const lastErasStartSessionIndexLookup = erasStartSessionIndexLookup.length - 1;
+
+    return erasStartSessionIndexLookup[lastErasStartSessionIndexLookup][0];
+  },
+  [erasStartSessionIndex, sessionEra, currentSessionMode]
+  );
+
+  useEffect(() => {
+    if (era) {
+      const sessionPeriod = Number(api.consts.elections.sessionPeriod.toString());
```

Comment:
nit: to readuce nesting: `if (!era) return;`

Reply:
Done